src/sources/soundsourcecoreaudio.cpp

```diff
@@ -115,9 +115,8 @@ SoundSource::OpenResult SoundSourceCoreAudio::tryOpen(
     m_bFileIsMp3 = m_inputFormat.mFormatID == kAudioFormatMPEGLayer3;
 
     // create the output format
-    const UInt32 numChannels =
-            params.getSignalInfo().getChannelCount().isValid()
-            ? params.getSignalInfo().getChannelCount()
+    const UInt32 numChannels = m_inputFormat.mChannelsPerFrame != 0
+            ? m_inputFormat.mChannelsPerFrame
```

Comment:
Do `params` contain bogus information here, should we fix that?

Reply:
`params` contains the maximum number of  supported channels. This used to always be `2` before STEMs were introduced.

Reply:
Ah hm, do the other sound sources not parse the channel count from `params`?

Reply:
No they should detect it from the file they are loading. I remember updating it (and forgot CoreAudio since it's on Mac only) but I'll give it a sanity check